### PR TITLE
Bac 148 add pagination control

### PR DESCRIPTION
**Pull request by nassery318**
npm run build is working
docker build . is working

### CODE REVIEW

File: frontend/src/app/[locale]/submit/page.tsx [hunk 1]
```diff
@@ -342,58 +346,91 @@ function SubmitPage() {
...
+                  <Typography>
+                    Click to upload your Outlook &apos;.msg&apos; files to
+                    Google Cloud Storage
+                  </Typography>
+                </>
+              }
+            >
+              <a
+                className="text-center underline rounded border shadow-sm p-[6px] hover:bg-gray-200"
+                aria-label='Click to upload your pdfs or Outlook msg files to Google Cloud Storage'
```
Comment (marcusmonteirodesouza) [inline frontend/src/app/[locale]/submit/page.tsx:387]:
```suggestion
                aria-label='Click to upload your Outlook &apos;.msg&apos; files to Google Cloud Storage'
```

File: frontend/src/app/[locale]/submit/page.tsx [hunk 2]
```diff
@@ -457,27 +494,41 @@ function SubmitPage() {
...
+              disabled={pageNumber === 1}
+              className={`bg-main-darkBlue rounded text-white flex items-center gap-2 px-2 ${pageNumber===1 ? 'disabled:bg-gray-400 cursor-not-allowed': ''}`}
+              aria-label="Previous Entries in the Table"
+            >
+              <MdArrowBack /> Previous entries
+            </button>
+            <button
+              type="button"
+              onClick={async () => await pageForward()}
+              disabled={pageNumber * pageSize >= totalCount}
```
Comment (marcusmonteirodesouza) [inline frontend/src/app/[locale]/submit/page.tsx:512]: Maybe create a function that returns if it's the last page?

File: frontend/src/app/[locale]/submit/page.tsx [hunk 3]
```diff
@@ -457,27 +494,41 @@ function SubmitPage() {
...
-          <p>
-            {pageNumber * pageSize - pageSize + 1} -{' '}
-            {Math.min(pageNumber * pageSize, totalCount)} of {totalCount}
+        {/* pagination */}
+        <div className="flex gap-4 p-2 justify-end border-b border-x rounded-b bg-gray-100 items-center">
+          <div className="flex flex-row gap-1">
+            <button
+              type="button"
+              onClick={async () => await pageBack()}
+              disabled={pageNumber === 1}
```
Comment (marcusmonteirodesouza) [inline frontend/src/app/[locale]/submit/page.tsx:503]: Maybe create a function to return if it's the first page?

File: frontend/src/app/[locale]/submit/page.tsx [hunk 4]
```diff
@@ -65,6 +68,7 @@ function SubmitPage() {
         [pageNumber + 1]: fetchDataExtractionsJson.nextPageToken,
       });
       setTotalCount(fetchDataExtractionsJson.count);
+      console.log('the total count is:', totalCount);
```
Comment (marcusmonteirodesouza) [inline frontend/src/app/[locale]/submit/page.tsx:71]: Should this be removed?

**Review by marcusmonteirodesouza — CHANGES_REQUESTED** [review]

File: frontend/src/app/[locale]/submit/page.tsx [hunk 5]
```diff
@@ -457,27 +494,41 @@ function SubmitPage() {
...
+              <MdArrowBack /> Previous entries
+            </button>
+            <button
+              type="button"
+              onClick={async () => await pageForward()}
+              disabled={pageNumber * pageSize >= totalCount}
+              className={`bg-main-darkBlue rounded text-white flex items-center gap-2 px-2 ${pageNumber * pageSize >= totalCount ? 'disabled:bg-gray-400 cursor-not-allowed': ''}`}
+              aria-label="Next Entries in the Table"
+            >
+              Next entries <MdArrowForward />
```
Comment (leechristensenYopp) [inline frontend/src/app/[locale]/submit/page.tsx:516]: Don't forget to translate

File: frontend/src/app/[locale]/submit/page.tsx [hunk 6]
```diff
@@ -457,27 +494,41 @@ function SubmitPage() {
...
+              aria-label="Previous Entries in the Table"
+            >
+              <MdArrowBack /> Previous entries
+            </button>
+            <button
+              type="button"
+              onClick={async () => await pageForward()}
+              disabled={pageNumber * pageSize >= totalCount}
+              className={`bg-main-darkBlue rounded text-white flex items-center gap-2 px-2 ${pageNumber * pageSize >= totalCount ? 'disabled:bg-gray-400 cursor-not-allowed': ''}`}
+              aria-label="Next Entries in the Table"
```
Comment (leechristensenYopp) [inline frontend/src/app/[locale]/submit/page.tsx:514]: Don't forget to translate

File: frontend/src/app/[locale]/submit/page.tsx [hunk 7]
```diff
@@ -457,27 +494,41 @@ function SubmitPage() {
...
-            {Math.min(pageNumber * pageSize, totalCount)} of {totalCount}
+        {/* pagination */}
+        <div className="flex gap-4 p-2 justify-end border-b border-x rounded-b bg-gray-100 items-center">
+          <div className="flex flex-row gap-1">
+            <button
+              type="button"
+              onClick={async () => await pageBack()}
+              disabled={pageNumber === 1}
+              className={`bg-main-darkBlue rounded text-white flex items-center gap-2 px-2 ${pageNumber===1 ? 'disabled:bg-gray-400 cursor-not-allowed': ''}`}
+              aria-label="Previous Entries in the Table"
```
Comment (leechristensenYopp) [inline frontend/src/app/[locale]/submit/page.tsx:505]: Don't forget to translate

File: frontend/src/app/[locale]/submit/page.tsx [hunk 8]
```diff
@@ -342,58 +346,91 @@ function SubmitPage() {
...
+                onChange={(ev) => handleSearchBarInputChange(ev.target.value)}
+                className="w-full"
+              />
+            </Paper>
+            <CustomTooltip
+              placement="top"
+              title={
+                <>
+                  <Typography>
+                    Click to upload your Outlook &apos;.msg&apos; files to
```
Comment (leechristensenYopp) [inline frontend/src/app/[locale]/submit/page.tsx:379]: Don't forget to translate

**Review by marcusmonteirodesouza — APPROVED** [review]
LGTM!

**Review by nassery318 — COMMENTED** [review]
comments addressed

File: frontend/src/app/[locale]/submit/page.tsx (hunk 1 above)
Comment (nassery318) [reply frontend/src/app/[locale]/submit/page.tsx]: Done!

**Pull request merged**